### PR TITLE
Fixing bug in properties iterator

### PR DESCRIPTION
**Pull request by escowles**
Fixes https://jira.duraspace.org/browse/FCREPO-1988

### CODE REVIEW

File: src/main/java/org/fcrepo/audit/InternalAuditor.java [hunk 1]
```diff
@@ -144,7 +144,7 @@ public void recordEvent(final FedoraEvent event) {
             isParentNodeLastModifiedEvent = true;
             final Iterator<String> propertiesIter = properties.iterator();
             String property;
-            while (properties.iterator().hasNext()) {
+            while (propertiesIter.hasNext()) {
```
Comment (ajs6f) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: While we're at it, I would find a filter expression clearer here, like:
```
isParentNodeLastModifiedEvent = false;
String property = iteratorToStream(properties.iterator()).filter(property -> 
 !property.equals(LAST_MODIFIED) && !property.equals(LAST_MODIFIED_BY)
).findFirst().orElse(() -> { isParentNodeLastModifiedEvent = true ; return someDefault; } );
```

Comment (acoburn) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: since `properties` is a `Set`, this would be better: `properties.stream().filter(...)`

Comment (ajs6f) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: Good point, @acoburn++.

**Comment by escowles** [comment]
@ajs6f @acoburn: I've updated the PR to address your comments, and update checkstyle to handle the Java 8 syntax.

File: pom.xml [hunk 2]
```diff
@@ -42,7 +42,7 @@
     <guava.version>18.0</guava.version>
     <slf4j.version>1.7.7</slf4j.version>
     <!-- plugin versions -->
-    <checkstyle.plugin.version>2.14</checkstyle.plugin.version>
+    <checkstyle.plugin.version>2.15</checkstyle.plugin.version>
```
Comment (acoburn) [inline pom.xml]: Actually, removing this line altogether would be a better option. The checkstyle version should be picked up from the fcrepo-parent pom.

File: src/main/java/org/fcrepo/audit/InternalAuditor.java [hunk 3]
```diff
@@ -139,22 +148,16 @@ public void recordEvent(final FedoraEvent event) {
...
-                }
-            }
+
+        /* adding/removing a file updates the lastModified property of the parent container,
+           so ignore updates when only lastModified is changed. */
+        if (eventType.contains(PROPERTY_CHANGED)) {
+            final String changedProperty = event.getProperties().stream().filter(p ->
+                !p.equals(LAST_MODIFIED) && !p.equals(LAST_MODIFIED_BY)
+            ).findFirst().orElse("");
+            isParentNodeLastModifiedEvent = changedProperty.equals("");
```
Comment (ajs6f) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: Nice, this is better/clearer than what I suggested. The other thing might be to leave `changedProperty` as an `Optional<>` and resolve it later, so you get `isParentNodeLastModifiedEvent = changedProperty.isPresent()`, but it's up to you whether that's any clearer. I tend to want to resolve `Optional` ASAP.

File: pom.xml (hunk 2 above)
Comment (ajs6f) [inline pom.xml]: +1

Comment (ajs6f) [inline pom.xml]: If we are going to call out a new version, maybe (putting on my scary rubber @awoods mask) there's a separate ticket here to note that. If we're removing the version, it seems fine as part of this.

**Comment by escowles** [comment]
I like using `isPresent()` better than checking for equality with an empty string (and then we can slim this down even further).

File: src/main/java/org/fcrepo/audit/InternalAuditor.java [hunk 4]
```diff
@@ -139,22 +148,15 @@ public void recordEvent(final FedoraEvent event) {
...
-                    break;
-                }
-            }
+
+        /* adding/removing a file updates the lastModified property of the parent container,
+           so ignore updates when only lastModified is changed. */
+        if (eventType.contains(PROPERTY_CHANGED)) {
+            isParentNodeLastModifiedEvent = !event.getProperties().stream().filter(p ->
+                !p.equals(LAST_MODIFIED) && !p.equals(LAST_MODIFIED_BY)
+            ).findFirst().isPresent();
```
Comment (ajs6f) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: Oh, shazam! I thought we actually needed the property itself or I would have suggested this at first! :)

Comment (acoburn) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: ok, here's an even shorter version: skip the `.filter(...).findFirst().isPresent()` and replace it with `.anyMatch(...)`

**Pull request merged**